frontend/src/components/PopOverComponent.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { Box, Typography, Button, Popover } from '@mui/material';
+
+const PopoverComponent = ({ open, handleClose, selectedPlaceProperties, anchorElRef }) => {
+  return (
+    <div ref={anchorElRef} style={{ display: 'none', position: 'absolute' }}>
+      <Popover
+        id="popover"
+        open={open}
+        onClose={handleClose}
+        anchorEl={anchorElRef.current}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'center',
+        }}
+      >
+        <Box>
+          <Typography variant="h6" component="h2">
+            Location Properties
+          </Typography>
+          <Typography variant="body1" component="div" sx={{ mt: 2 }}>
+            <div>
```

Comment:
Consider exporting substantial blocks of code within render functions, such as the Box below, into a constant component. For example:
```
const locationProperties = (
   <Box>
          <Typography variant="h6" component="h2">
            Location Properties
          </Typography>
          <Typography variant="body1" component="div" sx={{ mt: 2 }}>
            <div>
              Location: {selectedPlaceProperties && selectedPlaceProperties.location}
            </div>
            <div>
              Coordinates: {selectedPlaceProperties && selectedPlaceProperties.coordinates}
            </div>
            <div>
              Degrees: {selectedPlaceProperties && selectedPlaceProperties.degrees}
            </div>
          </Typography>

          <Button onClick={handleClose} sx={{ mt: 2 }}>
            Close
          </Button>
     </Box>
)
```

Subsequently, modify your return statement accordingly:
```
return (
...
{ locationProperties }
...
)
```